### ISSUE TITLE
Don't show subscribed ical calendars in the calendar list?

### ISSUE THREAD

**Issue by szaimen** [nextcloud/tasks#910]
I have created this issue so that we can discuss whether subscribed ical calendars should be shown in the calendar list or not. https://github.com/nextcloud/tasks/issues/908#issuecomment-596215513
IMO, they shouldn't get shown like it was before on 0.11.3 - at least if no task is existant in there.

**Comment by raimund-schluessler** [comment]
I am a bit surprised that subscribed calendars didn't show up with 0.11.x already. I don't have any, so I cannot tell (I could test it though 🙈).

However, every calendar which has the `VTODO` component set, will show up. I think this is the expected behaviour since a subscribed calendar with `VTODO` can contain tasks which are wanted to be shown.

**Comment by raimund-schluessler** [comment]
The correct solution would be that a subscribed calendar without tasks should not contain the `VTODO` component.

What we can do on our side probably is https://github.com/nextcloud/tasks/issues/84.

**Comment by szaimen** [comment]
> The correct solution would be that a subscribed calendar without tasks should not contain the `VTODO` component.

Yes, but the subscribed calendar seems to have it and I will not be able to delete it.

> What we can do on our side probably is #84.

Beeing able to hide such calendars sound like a good idea 👍
I imagine this as an option here where you can click on "hide calendar"
![image](https://user-images.githubusercontent.com/42591237/76167350-37444180-6166-11ea-8174-012020b29bd5.png)

If there are hidden calendars, could be shown as an additional field below "add list" where you could click on "show all lists" or "show hidden lists" and could make them visible again.
![image](https://user-images.githubusercontent.com/42591237/76167404-afab0280-6166-11ea-8164-6afa1b544a5f.png)

**Comment by raimund-schluessler** [comment]
Then let's merge this issue with #84.

**Comment by ltGuillaume** [comment]
I'm having the same issue (all my subscribed calendars from https://fixtur.es) are shown in Tasks, _but_ they do NOT have VTODO specified in the table `oc_calendars` under `components`.

So where do you get the info from that they even support VTODO's?

**Comment by raimund-schluessler** [comment]
@ltGuillaume would it be possible that you share the complete link of one of the subscriptions (if you want by email) so I can try to debug this?

**Comment by ltGuillaume** [comment]
@raimund-schluessler https://ics.fixtur.es/v2/league/formula-1.ics

**Comment by raimund-schluessler** [comment]
@ltGuillaume Subscribed calendars shouldn't be listes in `oc_calendars` afaik, at least they do not show up there on my instance. They are listed in `oc_calendarsubscriptions` for me and this table doesn't have a components column. Are you sure they are really in this table?

@georgehrke Is there a way to figure out if a subscribed calendar has VTODOS?

**Comment by ltGuillaume** [comment]
@raimund-schluessler You are totally right, very sorry. The subscribed calendars are indeed listed only in `oc_calendarsubscriptions` inside the database. However, these calendars are also listed in Tasks somehow. When I subscribe to a new calendar, it's also added to Tasks.

> Also, the server returns ["VEVENT", "VTODO"] as components of the subscribed calendars.

Which server do you mean, fixtur.es? How can I check that?

**Comment by raimund-schluessler** [comment]
> Which server do you mean, fixtur.es? How can I check that?

Nextcloud server, it returns all subscribed calendars and shows them as also containing VTODOs. Don't know where Nextcloud gets this information from.

**Comment by ltGuillaume** [comment]
@raimund-schluessler Then I would say I would get the same issues with DAVx5+ICSx5, together with OpenTasks on Android, but I don't: DAVx5 only shows VTODO for two of my own calendars, OpenTasks only shows those two calendars as well, while the subscribed calendars are not shown as VEVEN,VTODO. Now it _could_ be that DAVx5/ICSx5 already assumes subscribed calendars don't have VTODO, I'm not sure about that, or that the calendar components are queried in a different way than your Tasks app does (that would make sense).

I do know, however, that it's a recent issue, which could mean it came with either the update of Nextcloud 18.0.1/18.0.2, or - if relevant - the update of the Calendar app to 2.0.1/2.0.2.

**Comment by georgehrke** [comment]
> Nextcloud server, it returns all subscribed calendars and shows them as also containing VTODOs. Don't know where Nextcloud gets this information from.

That is hardcoded in the server at the moment:
https://github.com/nextcloud/server/blob/master/apps/dav/lib/CalDAV/CalDavBackend.php#L1894

What I'm currently wondering about: You seem to be using c-dav's `findAllCalendars`, you shouldn't be seeing subscriptions anyway. I will check why that happens.

**Comment by georgehrke** [comment]
Here we go: https://github.com/nextcloud/cdav-library/pull/272

**Comment by ltGuillaume** [comment]
> Here we go: [nextcloud/cdav-library#272](https://github.com/nextcloud/cdav-library/pull/272)

Well, not yet apparently...

**Comment by raimund-schluessler** [comment]
Subscriptions won't be shown anymore with #942, will come with the next release.
